Hide windows x86 random failures on CI

piponazo opened the pull request:
After few days investigating the random windows x86 failures on CI (see [link](https://github.com/Exiv2/exiv2/actions/runs/847548972)), I decided to give up since I could not reproduce them locally. 

Anyways, the investigation bringts some improvements that I list here:
- CMake: remove some useless custom targets.
- CMake: find python3 with cmake and use the executable found.
- CMake: Change default debug compiler flags on MSVC to speed up buids (with `/Ox /Zo`). 
- Conan: use the latest expat 2.3.0 also on windows (do not use anymore the custom expat 2.2.6 I created a long time ago). With the usage of the conan `cmake_find_package` generator we can simplify few things. 
- Test `tests/bugfixes/github/test_issue_1472.py` was failing on windows without a bash terminal. 
- CI: Win Matrix builds also running on `main` on push events.

Finally I decided to not run the tests on windows x86 architectures to hide the failures. I think we do not lose too much since Windows x86 is marginal for Exiv2 and furthermore, I could not reproduce the failures locally.

piponazo committed:
cmake: find python3 and use it in tests

piponazo committed:
cmake: remove obsolete targets

piponazo committed:
ci - Win builds without unicode + with warnings as errors

neheb commented:
I initially thought the initial loop conversion PR was the cause. But reverting it doesn't seem to do anything.

I don't have a MSVC setup either.

piponazo committed:
cmake - Use cmake -E to be able to run tests on different windows shells

piponazo committed:
fix python tests on powershell

piponazo committed:
BMFF python test - Output strings in different lines

piponazo committed:
conan - change generator to be able to use latest expat version

piponazo committed:
make win debug builds a bit faster

piponazo committed:
Ignore missing PDBs

piponazo committed:
mac - cmake_paths does not work directly on mac

piponazo committed:
win - Only run tests on x64

On CI, pythonTests fail more or less consistently in the configuration
Arch:x86, BuildType:Relase, Shared:ON. I tried to reproduce this locally,
but I could not.

Since Win-x86 is marginal for Exiv2, we disable it from CI until
somebody raise concerns about it.

piponazo committed:
ci - Win matrix jobs will also run on:push for main

The idea is to generate a conan cache on main that could be reused in
future PRs to main.

piponazo commented:
Note that I have updated the description with information about the changes done and the decision taken to not run the tests on Win x86.

neheb commented:
Works for me. x86 windows is a dying platform.

X86 Linux and BSD are already not a thing.
